server/main.lua

```diff
@@ -55,47 +40,74 @@ RegisterNetEvent('qb-vehiclekeys:server:setVehLockState', function(vehNetId, sta
     SetVehicleDoorsLocked(NetworkGetEntityFromNetworkId(vehNetId), state)
 end)
 
-lib.callback.register('qbx-vehiclekeys:server:getVehicleKeys', function(source)
-    local citizenid = exports.qbx_core:GetPlayer(source).PlayerData.citizenid
-    local keysList = {}
-    for plate, citizenids in pairs (vehicleList) do
-        if citizenids[citizenid] then
-            keysList[plate] = true
-        end
-    end
-    return keysList
+local function getCitizenId(source)
+    local player = exports.qbx_core:GetPlayer(source)
+
+    if not player then return end
+    local citizenid = player.PlayerData.citizenid
+
+    return citizenid
+end
+
+RegisterNetEvent('qbx-vehiclekeys:server:setPlayerKeys', function()
+    local src = source
+    local citizenid = getCitizenId(src)
+    Player(src).state.keysList = keysList[citizenid]
 end)
 
 -----------------------
 ----   Functions   ----
 -----------------------
-function GiveKeys(id, plate)
-    local citizenid = exports.qbx_core:GetPlayer(id).PlayerData.citizenid
 
-    if not vehicleList[plate] then vehicleList[plate] = {} end
-    vehicleList[plate][citizenid] = true
+-- local function isVehicleVisibleByServer(plate)
+--     local vehicles = GetAllVehicles()
+--     if type(vehicles) == "table" then
+--         for _, vehicle in ipairs(vehicles) do
+--             local p = GetVehicleNumberPlateText(vehicle)
+--             if p == plate then
+--                 Wait(500)
+--                 if GetEntityPopulationType(vehicle) == 7 then
+--                     return true
+--                 end
+--             end
+--         end
+--     end
+-- end
```

Comment:
can remove?

Reply:
Need a function that would check if a vehicle can disappear when player walks away from it. The vehicle will be ignored by keysList collection.